utt-replica/signature-processor/tests/sigProcessorTests.cpp

```diff
@@ -293,9 +293,8 @@ class test_utt_instance : public ::testing::Test {
       message_comms[i] = msc;
       sig_processors[i] = std::make_shared<utt::SigProcessor>(
           i, n, t, 1000, msc, msr, ((IncomingMsgsStorageImp*)(ims.get()))->timers());
-      auto sp = sig_processors[i];
-      msr->registerMsgHandler(MsgCode::ClientRequest, [&, sp](std::unique_ptr<bftEngine::impl::MessageBase> message) {
-        auto msg = std::make_unique<ClientRequestMsg>(message.release());
+      msr->registerMsgHandler(MsgCode::ClientRequest, [&, i](bftEngine::impl::MessageBase* message) {
+        ClientRequestMsg* msg = (ClientRequestMsg*)message;
```

Comment:
where do we use this handler?

Reply:
In a real system, replicaImp implements this method. In the test, we need a way to handle the "callback" from the signature collector. This is why we have this handler here